src/middlewares/auth.ts

```diff
@@ -0,0 +1,23 @@
+import { NextFunction, Request, Response } from 'express';
+
+import Firebase from 'src/config/firebase';
+
+export interface RequestWithFirebase extends Request {
+  firebaseUid?: string;
+}
+
+export const checkToken = async (req: RequestWithFirebase, res: Response, next: NextFunction) => {
+  if (!req.headers.authorization) {
+    return res.boom.badRequest('Token is required');
+  }
+
+  return Firebase.auth()
+    .verifyIdToken(req.headers.authorization)
+    .then((decodedToken) => {
+      req.firebaseUid = decodedToken.uid;
```

Comment:
`res.locals` is intended to be used this way, storing data which will be used by further middlewares/controllers
```suggestion
      res.locals.firebaseUid= decodedToken.uid;
```

Adding the role might be useful as well

Reply:
Fixed.